src/main/java/Duke.java

```diff
@@ -1,10 +1,157 @@
+                printList(list, tasks);
+            } else if (input.startsWith ("done")) {
+                setDone(input, list, tasks);
+            } else if (input.startsWith ("todo")) {
+                input = input.replace("todo ", "");
+                list.add(input);
+                tasks[count] = new ToDo(input);
+                count = printAddTaskMessage (count, list, tasks);
+            } else if (input.startsWith ("event")) {
+                input = input.replace("event ", "");
+                int positionIndex = input.indexOf("/");
+                String date = input.substring(positionIndex+1, input.length());
+                input = input.substring(0, positionIndex);
+                tasks[count] = new Event(input, date);
+                list.add(tasks[count].getDescription());
+                count = printAddTaskMessage (count, list, tasks);
+            } else if (input.startsWith ("deadline")) {
+                input = input.replace("deadline ", "");
+                int positionIndex = input.indexOf("/");
+                String date = input.substring(positionIndex + 1, input.length());
+                input = input.substring(0, positionIndex);
+                tasks[count] = new Deadline(input, date);
+                list.add(tasks[count].getDescription());
+                count = printAddTaskMessage (count, list, tasks);
+            } else {
+                System.out.println("Please input valid task description......");
+                printLine();
+            }
+            input = in.nextLine();
+        }
+        printBye();
+    }
+
+    private static int printAddTaskMessage (int count, ArrayList<String> list, Task[] tasks) {
+        System.out.println("Got it. I've added this task:" );
+        System.out.println("  ["
+                + tasks[count].getType()
+                + "]"
+                + "[" + tasks[count].getStatusIcon()
+                + "] "
+                + list.get(count));
+        System.out.println("Now you have " + list.size() +" tasks in the list.");
+        printLine();
+        count++;
+        return count;
+    }
+
+    private static void setDone(String input, ArrayList<String> list, Task[] tasks) {
+        if (list.size()==0) {
+            System.out.println("The list is empty");
+            printLine();
+        } else {
+            String inputNumber;
+            int taskNumber;
+            inputNumber = input.replaceAll("[^0-9]", "");//replace all non-number with space
+            taskNumber = Integer.parseInt(inputNumber);
+            if (taskNumber > list.size()) {
+                System.out.println("The task number is out of bound! Please type \"list\"");
+                printLine();
+            } else if (tasks[taskNumber-1].isDone==true){
+                System.out.println("Chill man, this task is completed!");
+                printLine();
+            } else {
+                tasks[taskNumber - 1].makeAsDone();
```

Comment:
Not a violation of coding standard but there's an inconsistency between [taskNumber-1] at line 100 and 
[taskNumber - 1] at line 104